lxc/list.go

```diff
@@ -281,16 +283,16 @@ func (c *listCmd) listContainers(d *lxd.Client, cinfos []shared.ContainerInfo, f
 	}
 
 	for _, cInfo := range cinfos {
-		cStatesLock.Lock()
-		cSnapshotsLock.Lock()
 		for _, column := range columns {
 			if column.NeedsState && cInfo.IsActive() {
 				_, ok := cStates[cInfo.Name]
 				if ok {
 					continue
 				}
 
+				cStatesLock.Lock()
 				cStates[cInfo.Name] = nil
+				cStatesLock.Unlock()
```

Comment:
Yeah, I understand now. The problem should occur every time with more containers than threads, in fact, because we never allow the threads to complete until we write all of the stuff to the states/snapshots queues. I suppose we could bump the queue size, but this works just as well for me.